sprokit/processes/adapters/embedded_pipeline.h

```diff
@@ -122,21 +122,35 @@ class KWIVER_ADAPTER_EXPORT embedded_pipeline
   embedded_pipeline();
   virtual ~embedded_pipeline();
 
+  /**
+   * @brief Set application information
+   *
+   * This method sets the application information that will be used when
+   * searching for pipeline files. See kwiver::vital::read_config_file for a
+   * detailed explanation of the parameters and how search paths work.
+   *
+   * \sa build_pipeline
+   */
+  void set_application_information( std::string const& app_name,
+                                    std::string const& app_version,
+                                    std::string const& app_prefix = {} );
+
   /**
    * @brief Build the embedded pipeline.
    *
-   * This method creates the pipeline based on the contents of the
-   * supplied stream.
+   * This method creates the pipeline based on the contents of the supplied
+   * stream. Inclusions will be resolved using the search paths in the same
+   * manner as kwiver::vital::read_config_file.
    *
    * @param istr Input stream containing the pipeline description.
    *
-   * @param def_dir The directory name used to report errors in the
-   * input stream and is used as the current directory to locate
-   * includes and to resolve relpath. Since the input stream being
-   * processed has no file name, the name "in-stream" is appended to
-   * the directory supplied so that errors in the stream can be
-   * differentiated from errors from other files. If this parameter is
-   * not supplied, the current directory is used.
+   * @param def_dir The directory name used to report errors in the input
+   * stream and is used as the current directory to locate includes (if not
```

Comment:
I believe files in the current directory take precedence over files on the config search path.  So it doesn't look in the current directory in not found in the config search path, instead it only uses the config search path if not found in the current directory.  Right?  Or does pipeline search work differently?

Reply:
The search paths take precedence ([implementation](https://github.com/Kitware/kwiver/blob/8b55a9c32aabb4b895f95d01666209d316a877d9/sprokit/src/sprokit/pipeline_util/lex_processor.cxx#L680-L731)), but note that the search paths also include `.` as the first entry. If a path can't be resolved via the search paths, the lexer then walks *the include stack*. Strictly speaking, searching the current directory is new behavior as of this PR (except when the include stack happens to include the current directory).

Reply:
Maybe I'm not following, but if `.` is the first entry on the search path and we look at the search path first then doesn't the current directory take precedence?  Searching the current directory again at the end wouldn't change that.  I know `.` is technically part of the search path, but that's not advertised to the user.  It's not in `KWIVER_CONFIG_PATH`.

Basically I'm ask this:  If I have `foo.pipe` installed in `config/pipelines/foo.pipe` and I also put a `foo.pipe` in my current working directory, which one is used?  And is the behavior different depending on whether `foo.pipe` is my top-level pipeline or if I `include foo.pipe`?

Reply:
Okay, I think I see the confusion. `.` is the current *working* directory, a la `getcwd()`. The *current directory*, as used in the context of this documentation, is the location of the pipeline "file". So if I run `./TheGui` from a shell and open `~/junk/sample.pipe`, `.` is (unless `setcwd()` has been used) the location of `TheGui`, while "current directory" as used in this documentation is `~/junk`. (Note also that "current directory" *changes* with each `include`, although what this documentation fails to mention is that the entire *stack* thereof is used, not just the top of said stack.)

That `.` is the first search path is documented in the referenced documentation of `read_config_file`, which I'd prefer to not duplicate in a half dozen places. (Not just here, also several other functions in [config_block_io.h](https://github.com/Kitware/kwiver/blob/master/vital/config/config_block_io.h).) Note also that there are quite a few paths in the default search paths besides what is in `KWIVER_CONFIG_PATH`.

>  If I have `foo.pipe` installed in `config/pipelines/foo.pipe` and I also put a `foo.pipe` in my current working directory, which one is used?

First off, the search wouldn't find both of those, because `<prefix>/config/pipelines` isn't in the search paths, only `<prefix>/config`. That said, assuming that you `include pipelines/foo.pipe` (the intended usage), that will prefer `./pipelines/foo.pipe` over `<prefix>/config/pipelines/foo.pipe`.

>  And is the behavior different depending on whether `foo.pipe` is my top-level pipeline or if I `include foo.pipe`?

In that *literal* example, I believe "no". In most respects, "no" in general; however, if you `include some/path/to/foo.pipe`, potentially yes, as any `include` in that `foo.pipe` will also try ("first", but only after the config search paths fail) paths relative to whatever `some/path/to` resolved into *and* ("second") relative to the location of your top-level pipe.

Reply:
I tweaked the documentation to hopefully make this more clear (and to fix the bad grammar).